pkgs/by-name/_6/_64gram/package.nix

```diff
@@ -19,19 +18,14 @@ telegram-desktop.overrideAttrs (old: rec {
     hash = "sha256-vRiAIGY3CU5+hsdn8xiNbgvSM3eGRVwnvsSmSoaDN/k=";
   };
 
-  patches = (old.patches or []) ++ [
-    (fetchpatch {
-      url = "https://github.com/TDesktop-x64/tdesktop/commit/c996ccc1561aed089c8b596f6ab3844335bbf1df.patch";
-      revert = true;
-      hash = "sha256-Hz7BXl5z4owe31l9Je3QOXT8FAyKcbsXsKjGfCmXhzE=";
-    })
-  ];
-
   cmakeFlags = (old.cmakeFlags or []) ++ [
     (lib.cmakeBool "DESKTOP_APP_DISABLE_AUTOUPDATE" true)
-    (lib.cmakeBool "disable_autoupdate" true)
   ];
 
+  env.NIX_CFLAGS_COMPILE = toString ((lib.splitString " " (old.env.NIX_CFLAGS_COMPILE or "")) ++ [
+    "-DTDESKTOP_DISABLE_AUTOUPDATE"
```

Comment:
Why is this still required after setting the cmake flags.

Reply:
Look at the commit changing this, it removes the patch uncommenting setting `-DTDESKTOP_DISABLE_AUTOUPDATE` on disable_autoupdate cmake option. It seemed to me it's better but if you disagree, I can remove this commit.

From a quick glance, it looks like 64gram ruined the cmake code handling the autoupdate option as with the patch it required both DESKTOP_APP_DISABLE_AUTOUPDATE and disable_autoupdate cmake options while without patch it needs DESKTOP_APP_DISABLE_AUTOUPDATE cmake option + manual setting of cflag.

Reply:
NIX_CFLAGS_COMPILE is a sledgehammer, I'd prefer stick with the patch.